Utils/src/PmssmProducer.cc

```diff
@@ -15,80 +15,109 @@
 #include <sstream>
 #include <iostream>
 #include <iomanip>
+#include <fstream>
 
 //
 // class declaration
 //
 
 class PmssmProducer : public edm::EDProducer {
-	public:
-		explicit PmssmProducer(const edm::ParameterSet&);
-		~PmssmProducer();
-		static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
-
-	private:
-		virtual void beginJob();
-		virtual void produce(edm::Event&, const edm::EventSetup&);
-		virtual void endJob();
+public:
+  explicit PmssmProducer(const edm::ParameterSet&);
+  ~PmssmProducer();
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  virtual void beginJob();
+  virtual void produce(edm::Event&, const edm::EventSetup&);
+  virtual void endJob();
 		
-		virtual void beginRun(edm::Run&, edm::EventSetup const&);
-		virtual void endRun(edm::Run&, edm::EventSetup const&);
-		virtual void beginLuminosityBlock(edm::LuminosityBlock&, edm::EventSetup const&);
-		virtual void endLuminosityBlock(edm::LuminosityBlock&, edm::EventSetup const&);
+  virtual void beginRun(edm::Run&, edm::EventSetup const&);
+  virtual void endRun(edm::Run&, edm::EventSetup const&);
+  virtual void beginLuminosityBlock(edm::LuminosityBlock&, edm::EventSetup const&);
+  virtual void endLuminosityBlock(edm::LuminosityBlock&, edm::EventSetup const&);
 		
-		void process(std::string line, char delim, std::vector<std::string>& fields);
+  void process(std::string line, char delim, std::vector<std::string>& fields);
+  double getXsec(double pmssmid);
 		
-		// ----------member data ---------------------------
-		edm::GetterOfProducts<LHEEventProduct> getterOfProducts_;
-		bool shouldScan_, debug_;
+  // ----------member data ---------------------------
+  edm::GetterOfProducts<LHEEventProduct> getterOfProducts_;
```

Comment:
Are the pMSSM samples still produced with the old-style LHE method, i.e. this producer doesn't need to access `GenLumiInfoHeader`?